qfdmo/models/action.py

```diff
@@ -110,89 +110,51 @@ def __str__(self):
-        cls._manage_cache_expiration()
-        if cls._cached_action_instances is None:
-            cls._cached_action_instances = list(
-                Action.objects.prefetch_related("directions", "groupe_action")
-            )
-        return cls._cached_action_instances
-
-    @classmethod
-    def get_groupe_action_instances(cls) -> QuerySet[GroupeAction]:
-        cls._manage_cache_expiration()
-        if cls._cached_groupe_action_instances is None:
-            cls._cached_groupe_action_instances = GroupeAction.objects.prefetch_related(
-                "actions"
-            ).order_by("order")
-        return cls._cached_groupe_action_instances
-
-    # TODO : to be factorized
-    @classmethod
-    def get_actions_by_direction(cls) -> dict:
-        cls._manage_cache_expiration()
-        if cls._cached_actions_by_direction is None:
-            cls._cached_actions_by_direction = {
-                d.code: sorted(
-                    [
-                        model_to_dict(a, exclude=["directions"])
-                        for a in d.actions.filter(afficher=True)
-                    ],
-                    key=lambda x: x["order"],
-                )
-                for d in ActionDirection.objects.all()
-            }
-
-        return cls._cached_actions_by_direction
-
-    @classmethod
-    def get_directions(cls, first_direction=None) -> List[dict]:
-        cls._manage_cache_expiration()
-        if cls._cached_direction is None:
-            directions = ActionDirection.objects.all()
-            directions_list = [model_to_dict(d) for d in directions]
-            sorted_directions = sorted(directions_list, key=lambda x: x["order"])
-            cls._cached_direction = sorted_directions
-        if first_direction is not None and first_direction in [
-            d["code"] for d in cls._cached_direction
-        ]:
-            return sorted(
-                cls._cached_direction,
-                key=lambda x: (x["code"] != first_direction, x["code"]),
-            )
-        return cls._cached_direction
-
-    @classmethod
-    def get_reparer_action_id(cls):
-        cls._manage_cache_expiration()
-        if cls._reparer_action_id is None:
-            cls._reparer_action_id = Action.objects.get(code="reparer").id
-        return cls._reparer_action_id
-
-    @classmethod
-    def reload_cache(cls):
-        cls._cached_actions_by_direction = None
-        cls._cached_direction = None
-        cls._reparer_action_id = None
-        cls._cached_action_instances = None
-        cls._cached_groupe_action_instances = None
-
-    @classmethod
-    def _manage_cache_expiration(cls):
-        if cls._last_cache_update is None:
-            cls._last_cache_update = time.time()
-            return
-
-        current_time = time.time()
-        if current_time - cls._last_cache_update > 300:
-            cls.reload_cache()
-            cls._last_cache_update = current_time
+def get_action_instances() -> List[Action]:
```

Comment:
Suggestion de renommage 

```suggestion
def list_all_action_instances() -> List[Action]:
```